logstash-core/lib/logstash/pipeline.rb

```diff
@@ -648,22 +586,13 @@ def shutdown_workers
   end
 
   # for backward compatibility in devutils for the rspec helpers, this method is not used
-  # in the pipeline anymore.
+  # anymore and just here to not break TestPipeline that inherits this class.
   def filter(event, &block)
-    maybe_setup_out_plugins
-    # filter_func returns all filtered events, including cancelled ones
-    filter_func(event).each {|e| block.call(e)}
   end
```

Comment:
@ph the problem is that removing this thing requires a new dev-utils release. It does this:

```rb

`  class TestPipeline < LogStash::Pipeline
    public :filter
    public :flush_filters
  end
```

this makes the whole thing very involved. I'd rather fix this in the next iteration, otherwise, we're introducing a whole new range of new issues from changing things up in dev-utils right?